<h2> CREAZIONE LISTA 7357 COMUNI </h2>

<p> Ricavo dal Dataframe dei decessi totale dal 2015 al 30 giugno 2020 le lista dei comuni presenti che sono 7357. </p>

In [1]:
import pandas as pd
df = pd.read_csv('csv/decessi_istat_7357comuni.csv')
df.head()

REG NOME_REGIONE  PROV NOME_PROVINCIA  COD_PROVCOM NOME_COMUNE        DATA  \
0    1     Piemonte     1         Torino         1001       Agliè  2018-01-02   
1    1     Piemonte     1         Torino         1001       Agliè  2016-01-04   
2    1     Piemonte     1         Torino         1001       Agliè  2020-01-05   
3    1     Piemonte     1         Torino         1001       Agliè  2015-01-06   
4    1     Piemonte     1         Torino         1001       Agliè  2018-01-06   

   TOTALE  
0       1  
1       1  
2       1  
3       1  
4       2

<p> Prendo in esame il periodo fino al 30 aprile per poter fare il confronto nello stesso periodo nel DataFrame con 7904 comuni. </p> 

In [2]:
df['DATA'] = pd.to_datetime(df['DATA'])
df_data = df[df['DATA'] < '2020-05-01']

In [3]:
comuni = df_data['COD_PROVCOM'].unique()
comuni

array([  1001,   1002,   1003, ..., 111105, 111106, 111107], dtype=int64)

In [4]:
len(comuni)

7357

<p> La lista <b> comuni </b> contiene i 7357 comuni presenti del DataFrame dal 2015 al 30 giugno 2020. </p>

<h2> RICERCA DEI COMUNI MANCANTI </h2>

In [5]:
df2 = pd.read_csv('csv/decessi.csv', encoding = "ISO-8859-1")
df2.head()

REG  PROV NOME_REGIONE NOME_PROVINCIA NOME_COMUNE  COD_PROVCOM  \
0    1     1     Piemonte         Torino       Agliè         1001   
1    1     1     Piemonte         Torino       Agliè         1001   
2    1     1     Piemonte         Torino       Agliè         1001   
3    1     1     Piemonte         Torino       Agliè         1001   
4    1     1     Piemonte         Torino       Agliè         1001   

   TIPO_COMUNE  CL_ETA   GE        Data Maschi Femmine Totale  
0            1       7  609  09-06-2015      0       0      0  
1            1       7  609  09-06-2016      0       0      0  
2            1       7  609  09-06-2017      0       0      0  
3            1       7  609  09-06-2018      0       0      0  
4            1       7  609  09-06-2019      1       0      1

In [6]:
#Numero totale di comuni 
len(df2['COD_PROVCOM'].unique())

7904

In [7]:
#Numero totale delle province
len(df2['NOME_PROVINCIA'].unique())

107

In [8]:
comuni_mancanti = df2[df2['COD_PROVCOM'].isin(comuni) == False]
comuni_mancanti.head()

REG  PROV NOME_REGIONE NOME_PROVINCIA NOME_COMUNE  COD_PROVCOM  \
17250    1     1     Piemonte         Torino   Balangero         1016   
17251    1     1     Piemonte         Torino   Balangero         1016   
17252    1     1     Piemonte         Torino   Balangero         1016   
17253    1     1     Piemonte         Torino   Balangero         1016   
17254    1     1     Piemonte         Torino   Balangero         1016   

       TIPO_COMUNE  CL_ETA   GE        Data Maschi Femmine Totale  
17250            2       6  821  21-08-2015      0       0      0  
17251            2       6  821  21-08-2016      0       0      0  
17252            2       6  821  21-08-2017      1       0      1  
17253            2       6  821  21-08-2018      0       0      0  
17254            2       6  821  21-08-2019      0       0      0

In [9]:
len(comuni_mancanti['COD_PROVCOM'].unique())

547

<p> Il DataFrame <b> comuni_mancanti </b> contiene tutti i comuni che non sono presenti nel DataFrame usato per l'analisi dei decessi.
    <br />
    Nello specifico, si tratta di <b>547</b> comuni.
</p>

<h3> Modifica del DataFrame comuni_mancanti </h3>

In [10]:
#elimino le colonne che al momento non ci interessano
comuni_mancanti.drop(['REG', 'NOME_REGIONE', 'TIPO_COMUNE','CL_ETA', 'GE', 'Maschi', 'Femmine', ' Data', 'Maschi', 
                      'Femmine', 'Totale'], axis=1, inplace=True)
comuni_mancanti.head()

c:\users\monta\appdata\local\programs\python\python38\lib\site-packages\pandas\core\frame.py:3990: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  return super().drop(


PROV NOME_PROVINCIA NOME_COMUNE  COD_PROVCOM
17250     1         Torino   Balangero         1016
17251     1         Torino   Balangero         1016
17252     1         Torino   Balangero         1016
17253     1         Torino   Balangero         1016
17254     1         Torino   Balangero         1016

In [11]:
len(comuni_mancanti['NOME_COMUNE'].unique())

547

In [12]:
# salvo il DataFrame
comuni_mancanti.to_csv('csv/comuni_province_mancanti.csv', index=False)

<h3> RICERCA DELLE PROVINCE DI CUI NON ABBIAMO I DATI DI TUTTI I COMUNI CHE NE FANNO PARTE </h3>

In [13]:
lista_province = comuni_mancanti['NOME_PROVINCIA'].unique()
lista_province

array(['Torino', 'Vercelli', 'Novara', 'Cuneo', 'Asti', 'Alessandria',
       "Valle d'Aosta/Vallée d'Aoste", 'Imperia', 'Savona', 'Genova',
       'La Spezia', 'Varese', 'Como', 'Milano', 'Bergamo', 'Brescia',
       'Pavia', 'Cremona', 'Mantova', 'Bolzano/Bozen', 'Trento', 'Verona',
       'Vicenza', 'Belluno', 'Treviso', 'Venezia', 'Padova', 'Udine',
       'Parma', "Reggio nell'Emilia", 'Modena', 'Bologna', 'Ferrara',
       'Forlì-Cesena', 'Pesaro e Urbino', 'Ancona', 'Macerata',
       'Ascoli Piceno', 'Massa-Carrara', 'Lucca', 'Pistoia', 'Firenze',
       'Livorno', 'Pisa', 'Arezzo', 'Siena', 'Grosseto', 'Perugia',
       'Terni', 'Viterbo', 'Rieti', 'Roma', 'Latina', 'Frosinone',
       'Caserta', 'Benevento', 'Napoli', 'Avellino', 'Salerno',
       "L'Aquila", 'Teramo', 'Pescara', 'Chieti', 'Campobasso', 'Foggia',
       'Bari', 'Taranto', 'Brindisi', 'Lecce', 'Potenza', 'Matera',
       'Cosenza', 'Catanzaro', 'Reggio Calabria', 'Trapani', 'Palermo',
       'Messina', 'Agrige

In [14]:
len(lista_province)

99

<p>La lista <b>lista_province</b> contiente il nome di tutte le province di cui non abbiamo i dati completi per ogni comune. Le province in questione risultano essere <b>99</b>. </p>

<h2> RICERCA DELLE PROVINCE COMPLETE DI TUTTI I COMUNI </h2>

In [15]:
tot_province = df2[df2['NOME_PROVINCIA'].isin(lista_province) == False]
tot_province.head()

REG  PROV NOME_REGIONE NOME_PROVINCIA             NOME_COMUNE  \
1905426    3    14    Lombardia        Sondrio  Albaredo per San Marco   
1905427    3    14    Lombardia        Sondrio  Albaredo per San Marco   
1905428    3    14    Lombardia        Sondrio  Albaredo per San Marco   
1905429    3    14    Lombardia        Sondrio  Albaredo per San Marco   
1905430    3    14    Lombardia        Sondrio  Albaredo per San Marco   

         COD_PROVCOM  TIPO_COMUNE  CL_ETA   GE        Data Maschi Femmine  \
1905426        14001            1      11  219  19-02-2015      0       0   
1905427        14001            1      11  219  19-02-2016      0       0   
1905428        14001            1      11  219  19-02-2017      0       0   
1905429        14001            1      11  219  19-02-2018      0       0   
1905430        14001            1      11  219  19-02-2019      0       1   

        Totale  
1905426      0  
1905427      0  
1905428      0  
1905429      0  
1905430      1

In [16]:
tot_province = tot_province['NOME_PROVINCIA'].unique()
tot_province

array(['Sondrio', 'Rovigo', 'Gorizia', 'Trieste', 'Piacenza', 'Ravenna',
       'Ragusa', 'Pordenone'], dtype=object)

<h2> CONCLUSIONE </h2>

<p> Le province di cui non possediamo i dati di tutti i comuni sono <b>99</b>. Per cui, andandole a eliminare dai dati della Protezione Civile che comprendono <b> 107 </b> province, risulta che le province per cui abbiamo i dati completi per ogni comune appartenente sono <b> 8 </b>.
    </p>

<h1> CREAZIONE DELLA NUOVA SERIE STORICA </h1>

<p> Nella nuova serie storica sono presenti i decessi dal 2015 al 30 giugno 2020 dei comuni che fanno parte della provincia di <b>Piacenza</b>, <b>Ravenna</b>, <b>Gorizia</b>, <b>Trieste</b>, <b>Pordenone</b>, <b>Sondrio</b>, <b>Ragusa</b> e <b>Rovigo</b>.

In [20]:
df_comuni_completi = df[df['NOME_PROVINCIA'].isin(tot_province)]
df_comuni_completi

REG           NOME_REGIONE  PROV NOME_PROVINCIA  COD_PROVCOM  \
282858     3              Lombardia    14        Sondrio        14001   
282859     3              Lombardia    14        Sondrio        14001   
282860     3              Lombardia    14        Sondrio        14001   
282861     3              Lombardia    14        Sondrio        14001   
282862     3              Lombardia    14        Sondrio        14001   
...      ...                    ...   ...            ...          ...   
1889755    6  Friuli-Venezia Giulia    93      Pordenone        93053   
1889756    6  Friuli-Venezia Giulia    93      Pordenone        93053   
1889757    6  Friuli-Venezia Giulia    93      Pordenone        93053   
1889758    6  Friuli-Venezia Giulia    93      Pordenone        93053   
1889759    6  Friuli-Venezia Giulia    93      Pordenone        93053   

                    NOME_COMUNE       DATA  TOTALE  
282858   Albaredo per San Marco 2018-01-15       1  
282859   Albaredo per San Marco 2018-01-20       1  
282860   Albaredo per San Marco 2020-01-29       1  
282861   Albaredo per San Marco 2015-01-30       1  
282862   Albaredo per San Marco 2017-02-15       1  
...                         ...        ...     ...  
1889755        Valvasone Arzene 2017-12-24       1  
1889756        Valvasone Arzene 2015-12-27       1  
1889757        Valvasone Arzene 2017-12-27       1  
1889758        Valvasone Arzene 2018-12-29       1  
1889759        Valvasone Arzene 2019-12-31       1  

[87108 rows x 8 columns]

In [21]:
# salvo il DataFrame
df_comuni_completi.to_csv('csv/df_comuni_completi.csv', index=False)